geomesa-gs-wfs/src/main/scala/org/geomesa/gs/wfs/output/ArrowOutputFormat.scala

```diff
@@ -150,6 +136,45 @@ class ArrowOutputFormat(geoServer: GeoServer)
       hints.put(FLIP_AXIS_ORDER, java.lang.Boolean.valueOf(option.toString))
     }
   }
+
+  private def isPreSorted(request: GetFeatureRequest, index: Int, sortField: Option[String], sortReverse: Option[Boolean]): Boolean =
+    (for (field <- sortField; reverse <- sortReverse.orElse(Some(false))) yield {
+      request.getQueries.get(index).getSortBy match {
+        case list if list.size == 1 =>
+          val sort = list.get(0)
+          Option(sort.getPropertyName).exists(_.getPropertyName == field) &&
+            (sort.getSortOrder == SortOrder.DESCENDING) == reverse
+        case _ => false
+      }
+    }).getOrElse(false)
+
+  private def arrowVisitor(request: GetFeatureRequest, hints: Hints, index: Int,
+                           os: OutputStream, features: CloseableIterator[SimpleFeature], sft: SimpleFeatureType): Unit = {
+    val encoding = SimpleFeatureEncoding.min(hints.isArrowIncludeFid, hints.isArrowProxyFid, hints.isFlipAxisOrder)
+    val version = hints.getArrowFormatVersion.getOrElse(FormatVersion.ArrowFormatVersion.get)
+    val dictionaries = hints.getArrowDictionaryFields
+    val sortField = hints.getArrowSort.map(_._1)
+    val sortReverse = hints.getArrowSort.map(_._2)
+    val preSorted = isPreSorted(request, index, sortField, sortReverse)
+    val batchSize = hints.getArrowBatchSize.getOrElse(ArrowProperties.BatchSize.get.toInt)
+    val flattenStruct = hints.isArrowFlatten
+
+    val visitor = new ArrowVisitor(sft, encoding, version, dictionaries, sortField, sortReverse, preSorted, batchSize, flattenStruct)
+    features.foreach(visitor.visit)
+    visitor.getResult().results.asScala.foreach(os.write)
+  }
+
+  private def toByteArray(feature: SimpleFeature): Array[Byte] =
+    feature.getAttribute(0).asInstanceOf[Array[Byte]]
+
+  private def featureSequenceInputStream(iter: CloseableIterator[SimpleFeature]): SequenceInputStream = {
+    val streamEnumeration = new util.Enumeration[ByteArrayInputStream] {
+      private val iterator = iter.map { feature => new ByteArrayInputStream(toByteArray(feature)) }
+      override def hasMoreElements: Boolean = iterator.hasNext
+      override def nextElement(): ByteArrayInputStream = iterator.next()
+    }
+    new SequenceInputStream(streamEnumeration)
```

Comment:
a simpler way to do this is:
```suggestion
    val streamEnumeration = iter.map(f => new ByteArrayInputStream(toByteArray(f))).asJavaEnumeration
    new SequenceInputStream(streamEnumeration)
```

Reply:
good catch, will update it